netbox/circuits/models/virtual_circuits.py

```diff
@@ -0,0 +1,164 @@
+    )
+
+    clone_fields = (
+        'provider_network', 'provider_account', 'status', 'tenant', 'description',
+    )
+    prerequisite_models = (
+        'circuits.ProviderNetwork',
+    )
+
+    class Meta:
+        ordering = ['provider_network', 'provider_account', 'cid']
+        constraints = (
+            models.UniqueConstraint(
+                fields=('provider_network', 'cid'),
+                name='%(app_label)s_%(class)s_unique_provider_network_cid'
+            ),
+            models.UniqueConstraint(
+                fields=('provider_account', 'cid'),
+                name='%(app_label)s_%(class)s_unique_provideraccount_cid'
+            ),
+        )
+        verbose_name = _('virtual circuit')
+        verbose_name_plural = _('virtual circuits')
+
+    def __str__(self):
+        return self.cid
+
+    def get_status_color(self):
+        return CircuitStatusChoices.colors.get(self.status)
+
+    def clean(self):
+        super().clean()
+
+        if self.provider_account and self.provider_network.provider != self.provider_account.provider:
+            raise ValidationError({
+                'provider_account': "The assigned account must belong to the provider of the assigned network."
+            })
+
+    @property
+    def provider(self):
+        return self.provider_network.provider
+
+
+class VirtualCircuitTermination(
+    CustomFieldsMixin,
+    CustomLinksMixin,
+    TagsMixin,
+    ChangeLoggedModel
+):
+    virtual_circuit = models.ForeignKey(
+        to='circuits.VirtualCircuit',
+        on_delete=models.CASCADE,
+        related_name='terminations'
+    )
+    role = models.CharField(
+        verbose_name=_('role'),
+        max_length=50,
+        choices=VirtualCircuitTerminationRoleChoices,
+        default=VirtualCircuitTerminationRoleChoices.ROLE_PEER
+    )
+    interface = models.OneToOneField(
+        to='dcim.Interface',
+        on_delete=models.CASCADE,
+        related_name='virtual_circuit_termination'
+    )
+    description = models.CharField(
+        verbose_name=_('description'),
+        max_length=200,
+        blank=True
+    )
+
+    class Meta:
+        ordering = ['virtual_circuit', 'role', 'pk']
+        verbose_name = _('virtual circuit termination')
+        verbose_name_plural = _('virtual circuit terminations')
+
+    def __str__(self):
+        return f'{self.virtual_circuit}: {self.get_role_display()} termination'
+
+    def get_absolute_url(self):
+        return reverse('circuits:virtualcircuittermination', args=[self.pk])
+
+    def get_role_color(self):
+        return VirtualCircuitTerminationRoleChoices.colors.get(self.role)
+
+    def to_objectchange(self, action):
+        objectchange = super().to_objectchange(action)
+        objectchange.related_object = self.virtual_circuit
+        return objectchange
+
+    @property
+    def parent_object(self):
+        return self.virtual_circuit
+
+    @cached_property
+    def peer_terminations(self):
```

Comment:
Noting that this is a `@property` whereas in `CircuitTermination` it's a method, `get_peer_terminations`. I prefer the property style just for syntactic reasons but I'm wondering if it would be better to make the usage consistent in both places?